lib/fog/storage/google_json/requests/copy_object.rb

```diff
@@ -15,9 +15,19 @@ def copy_object(source_bucket, source_object,
                         target_bucket, target_object, options = {})
           request_options = ::Google::Apis::RequestOptions.default.merge(options)
 
+          object = ::Google::Apis::StorageV1::Object.new(options)
+
           @storage_json.copy_object(source_bucket, source_object,
                                     target_bucket, target_object,
-                                    request_options, **options)
+                                    object, options: request_options, **filter_keyword_args(options))
```

Comment:
`request_options` being passed as 5th argument here was just wrong, it is expected to be a `Google::Apis::StorageV1::Object` according to the google-api-ruby-client.
https://github.com/googleapis/google-api-ruby-client/blob/abb3c968fecb1d62384b75811f20a703bcacd1e7/generated/google-apis-storage_v1/lib/google/apis/storage_v1/service.rb#L1682